src/libraries/Native/Unix/System.Native/pal_signal.c

```diff
@@ -154,13 +222,25 @@ static void* SignalHandlerLoop(void* arg)
         }
         else if (signalCode == SIGCONT)
         {
+            // TODO: should this be cancelable?
```

Comment:
Currently the `Console` stuff isn't cancelable.

If we want that, we should also make the `ConfigureTerminalForChildProcess` call that happens on SIGCHLD cancelable.

Reply:
All `Console` stuff would have to be cancelable for my [System.Terminal](https://github.com/alexrp/system-terminal) project to be able to rely on this API, fwiw. I currently rely on Mono.Posix replacing the runtime's signal handlers to prevent it from overriding `termios` state changes I've made.

Reply:
I've pushed a change that allows to `Cancel` the terminal configuration that happens on SIGCHLD/SIGCONT.